source/MaterialXGenShader/Syntax.h

```diff
@@ -66,9 +66,23 @@ class Syntax
 
     /// Returns a type qualifier to be used when declaring types for output variables.
     /// Default implementation returns empty string and derived syntax classes should
-    /// override  this method.
+    /// override this method.
     virtual const string& getOutputQualifier() const { return EMPTY_STRING; };
 
+    /// Get the qualifier used when declaring constant variables.
+    /// Default implementation returns empty string and derived syntax classes should
+    /// override this method.
+    virtual const string& getConstantQualifier() const { return EMPTY_STRING; };
```

Comment:
Maybe this should be a pure virtual method, to force derived classes to specify a constant qualifier. Just so this is not missed when adding support for a new language. I guess the same thing goes for output and uniform qualifiers, they could all be pure virtual.